### PR TITLE
2326: Types of property '{0}' are incompatible.

### PR DESCRIPTION
**Pull request by eddyw**
# Description

Screen Shot 2022-05-07 at 01 14 41

### CODE REVIEW

File: packages/engine/errors/2326.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+original: "Types of property '{0}' are incompatible."
+excerpt: "Two object types that look-alike have a property '{0}' with a different type."
```
Comment (mattpocock) [inline packages/engine/errors/2326.md:3]:
```suggestion
excerpt: "Two similar objects have a different '{0}' property."
```

Comment (mattpocock) [reply packages/engine/errors/2326.md:3]: Suggested, can merge without this.

**Review by mattpocock — APPROVED** [review]
Approved, with a comment for you to merge or disagree with. Nice explanation.

File: packages/engine/errors/2326.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+You're assigning to a variable whose type `A` matches `B` but they both have a property whose type is incompatible with each other. For example:
+
+```ts
+interface Foo {
+  {0}: string
+}
+interface Bar {
+  {0}: number
+}
+const foo: Foo = { value: "" }
```
Comment (eddyw) [inline packages/engine/errors/2326.md:15]: damn, this is wrong:
```suggestion
const foo: Foo = { {0}: "" }
```

File: packages/engine/errors/2326.md (hunk 1 above)
Comment (eddyw) [reply packages/engine/errors/2326.md:3]: See my comment: https://github.com/mattpocock/ts-error-translator/pull/69/files#r865676642

So, it's not that 2 similar objects have a **different property** `{0}`. It's that those 2 object types have **the same property**, the only difference is that the property's type is different in those 2 object types. e.g: `{ foo: string }` and `{ foo: number }` are structurally compatible (same keys) but the types of property `foo` are not assignable to each other `string <-> number`. This is always followed by another error that explains exactly that `string <-> number` so in this particular error I don't want to repeat that but only mention that this exact error means your objects share the same props but one of them has a different type that makes your objects not compatible.

Hm, I could maybe phrase it like this?
```
excerpt: "Two similar object have the same property '{0}' with different type."
```
Not sure if this is better 😓

File: packages/engine/errors/2326.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+original: "Types of property '{0}' are incompatible."
+excerpt: "Two similar types have a property '{0}' with a different type."
```
Comment (mattpocock) [inline packages/engine/errors/2326.md:3]:
```suggestion
excerpt: "Two similar types have a property '{0}' which is different."
```

**Pull request merged**